src/windowmanagement/TaskBar.tsx

```diff
@@ -15,29 +16,51 @@ export function TaskBar({ addWindow, visibleMenuItems }: TaskBarProps) {
   });
 
   return (
-    <Group
+    <Flex
+      gap={2}
       style={{
         backgroundColor: '#00000080',
-        height: 60,
-        paddingLeft: 'var(--mantine-spacing-md)'
+        whiteSpace: 'nowrap',
+        overflowX: 'auto',
+        overflowY: 'hidden'
+      }}
+      onWheel={(event) => {
+        event.currentTarget.scrollLeft += event.deltaY;
       }}
     >
       {checkedMenuItems.map((item) => {
-        const handleClick = () =>
+        function handleClick() {
           addWindow(item.content, {
             title: item.title,
             position: item.preferredPosition,
             id: item.componentID
           });
+        }
```

Comment:
Place this function in the component body instead? And pass item as an argument? That way we don't have to redefine the function each render